app/uk/gov/hmrc/eusubsidycompliancefrontend/controllers/SubsidyController.scala

```diff
@@ -100,45 +100,40 @@ class SubsidyController @Inject()(
 
   def getClaimAmount: Action[AnyContent] = escAuthentication.async { implicit request =>
     implicit val eori: EORI = request.eoriNumber
-    //TODO add 'getPrevious to all'
     store.get[SubsidyJourney].flatMap {
       case Some(journey) =>
-        journeyTraverseService.getPrevious[SubsidyJourney].flatMap { previous =>
-          journey
-            .claimAmount
-            .value
-            .fold(
-              Future.successful(
-                Ok(addClaimAmountPage(claimAmountForm, previous))
-              )
-            ) { x =>
-              Future.successful(
-                Ok(addClaimAmountPage(claimAmountForm.fill(x), previous))
-              )
-            }
-
+        val form = journey.claimAmount.value.fold(claimAmountForm)(claimAmountForm.fill)
+        for {
+          previous <- journeyTraverseService.getPrevious[SubsidyJourney]
```

Comment:
NABD but I'm seeing two patterns in getting the previous in this file. Flat map and for-comprehension. Can we stick to one?

Reply:
NABD,  i can change that

Reply:
Might be an idea to hold off these refactors of SubsidyController. I've got some changes that are nearly ready to go that clean these up. I've introduced OptionT to clean things up in a couple of places.